internal/expr/parser.go

```diff
@@ -109,6 +109,62 @@ func (p *Parser) add(part queryPart) {
 	p.partStart = p.pos
 }
 
+// removeComments passes over the input removing comments.
+// It does not allow for nested comments.
+func (p *Parser) removeComments() error {
+	var res bytes.Buffer
+	var textStart = p.pos
+	var textEnd = p.pos
+
+	for p.pos < len(p.input) {
+		textEnd = p.pos
+		// Ignore comment chars in string literals.
+		if ok, err := p.skipStringLiteral(); err != nil {
+			return err
+		} else if ok {
+			continue
+		}
+
+		if p.skipByte('-') || p.skipByte('/') {
+			startPos := p.pos - 1
+			c := p.input[startPos]
+			var end byte
+			if (c == '-' && p.skipByte('-')) || (c == '/' && p.skipByte('*')) {
+				if c == '-' {
+					end = '\n'
+				} else {
+					end = '*'
+				}
+				for p.pos < len(p.input) {
+					if p.input[p.pos] == end {
+						// if end == '\n' dont p.pos++ to keep the newline.
+						if end == '*' {
+							p.pos++
+							if !p.skipByte('/') {
+								continue
+							}
+						}
+						break
+					}
+					p.pos++
+				}
+
```

Comment:
If the end character for the comment part is never found, this may be an issue for `p.init()` later on, as the whole content starting from the start of the comment which never ends until the end of input is skipped, perhaps removing parts of the statement along with it. Maybe this should be treated as an error, and we only do `p.init()` below after clean parsing of the comments found? I think this issue applies to multi-line comments. Once this is handled, a test case could be added to check that such user error doesn't pass through undetected.

Reply:
In the sqlite spec it defines the comment as ending at the end char or the end of input. So if there is no end character then the comment spans to the end of input.